client/go/outline/parse.go

```diff
@@ -21,21 +21,32 @@ import (
 
 	"github.com/Jigsaw-Code/outline-apps/client/go/outline/platerrors"
 	"github.com/goccy/go-yaml"
-	"github.com/goccy/go-yaml/ast"
 )
 
-type parseTunnelConfigRequest struct {
-	Transport ast.Node
-	Error     *struct {
+// providerConfig is the config fetched from the provider. It may be either an error, or a tunnel config.
+type providerConfig struct {
+	ProviderErrorConfig  `yaml:",inline"`
+	ProviderTunnelConfig `yaml:",inline"`
+}
+
+// ProviderErrorConfig is config returned by the provider with a custom error to
+// show to the user.
+type ProviderErrorConfig struct {
+	Error *struct {
 		Message string
 		Details string
 	}
 }
 
-// tunnelConfigJson must match the definition in config.ts.
-type tunnelConfigJson struct {
-	FirstHop  string `json:"firstHop"`
-	Transport string `json:"transport"`
+// ProviderTunnelConfig is the config to fully configure the VPN.
```

Comment:
Can we have a bit more clarity on what should be included in the provider config? For instance, I implied earlier that clientconfig might include the reporting config, but reading this struct, it sounds like reporting config should go to the provider config?

Reply:
Reporting should go into the ClientConfig. But notice that the Client config is inline in the provider config, so the field will be top-level.